src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -540,5 +544,25 @@ await output.Conditions.EvaluateConditionsAsync(lazyInput, context, cancellation
 
             return state;
         }
+
+        private string DefineMutexKey(Message message, Flow flow, Identity userIdentity)
+        {
+            string parent = flow.Id;
+            string identity = userIdentity.ToString();
+
+            if (message.Metadata != null && message.Metadata.ContainsKey(TUNNEL_OWNER_METADATA))
```

Comment:
Should it check for `TUNNEL_ORIGINATOR_METADATA` too? I understand it's expected to have both or none, but is it possible a malformed message would have only one?